### PR TITLE
CB-7517 Check all taggings

### PR DESCRIPTION
**Pull request by Bajzathd**

### CODE REVIEW

File: common-model/src/main/java/com/sequenceiq/common/api/tag/base/TagsBase.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public class TagsBase implements Serializable {
+
+    @JsonValue
+    private final Map<String, String> values;
```
Comment (bergerdenes) [inline common-model/src/main/java/com/sequenceiq/common/api/tag/base/TagsBase.java:15]: call it `tags`

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/DatabaseStack.java [hunk 2]
```diff
@@ -12,12 +10,12 @@
...
 
-    private final Map<String, String> tags;
+    private final Tags tags;
 
-    public DatabaseStack(Network network, DatabaseServer databaseServer, Map<String, String> tags, String template) {
+    public DatabaseStack(Network network, DatabaseServer databaseServer, Tags tags, String template) {
         this.network = network;
         this.databaseServer = databaseServer;
-        this.tags = ImmutableMap.copyOf(tags);
+        this.tags = tags;
```
Comment (bergerdenes) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/DatabaseStack.java:18]: should it be the copy of tags with the copy constructor?

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudStack.java [hunk 3]
```diff
@@ -27,19 +28,19 @@
...
 
-    public CloudStack(Collection<Group> groups, Network network, Image image, Map<String, String> parameters, Map<String, String> tags, String template,
+    public CloudStack(Collection<Group> groups, Network network, Image image, Map<String, String> parameters, Tags tags, String template,
             InstanceAuthentication instanceAuthentication, String loginUserName, String publicKey, SpiFileSystem fileSystem) {
         this.groups = ImmutableList.copyOf(groups);
         this.network = network;
         this.image = image;
         this.parameters = ImmutableMap.copyOf(parameters);
-        this.tags = ImmutableMap.copyOf(tags);
+        this.tags = tags;
```
Comment (bergerdenes) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudStack.java:43]: should it be the copy of tags with the copy constructor?

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java [hunk 4]
```diff
@@ -230,11 +231,10 @@ private CreateFreeIpaRequest createFreeIpaRequest(EnvironmentDto environment) {
...
     private void setTags(EnvironmentDto environment, CreateFreeIpaRequest createFreeIpaRequest) {
-        Map<String, String> userDefinedTags = environment.getTags().getUserDefinedTags();
-        if (userDefinedTags == null) {
-            userDefinedTags = new HashMap<>();
-        }
-        createFreeIpaRequest.setTags(userDefinedTags);
+        Optional.ofNullable(environment.getTags())
+                .map(EnvironmentTags::getUserDefinedTags)
+                .map(tags -> new TagsRequest(tags.getAll()))
+                .ifPresent(createFreeIpaRequest::setTags);
```
Comment (bergerdenes) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java:237]: it won't set the empty Tags if there was no `userDefinedTags` present

File: integration-test/src/test/java/com/sequenceiq/it/cloudbreak/testcase/e2e/UniqueTestMethodNameTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                .collect(Collectors.groupingBy(Method::getName));
+
+        String errorMessage = methodsByName.entrySet()
+                .stream()
+                .filter(methodsEntry -> methodsEntry.getValue().size() > 1)
+                .map(methodsEntry -> getErrorMessage(methodsEntry.getKey(), methodsEntry.getValue()))
+                .collect(Collectors.joining(System.lineSeparator()));
+
+        if (!errorMessage.isEmpty()) {
+            Assert.fail(errorMessage);
```
Comment (bergerdenes) [inline integration-test/src/test/java/com/sequenceiq/it/cloudbreak/testcase/e2e/UniqueTestMethodNameTest.java:33]: fail with TestNG

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java (hunk 4 above)
Comment (Bajzathd) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java:237]: the request has an empty tags default value, so it is not necessary

**Comment by Bajzathd** [comment]
retest this please

**Comment by Bajzathd** [comment]
Merged in 3 separate PRs

**Pull request closed without merging**